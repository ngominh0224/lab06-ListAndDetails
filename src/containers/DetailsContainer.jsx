import React, { useState, useEffect } from 'react';
import CharacterList from '../components/characters/CharacterList';
import { fetchSingleCharacter } from '../services/heyArnoldApi';
import { useParams } from 'react-router';

const DetailsContainer = () => {
  const [loading, setLoading] = useState(true);
  const [character, setCharacter] = useState([]);
  const params = useParams();

  useEffect(() => {
    fetchSingleCharacter(params.id)
      .then(setCharacter)
      .finally(() => setLoading(false));
  }, []);

  if (loading) return <h1>Loading...</h1>;
  return (
    <div>
      <CharacterList characters={character} />
    </div>
  );
};

export default DetailsContainer;
