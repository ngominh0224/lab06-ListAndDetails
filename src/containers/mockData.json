[
  {
    "_id": "5da237699734fdcb7bef8f51",
    "name": "Arnold Shortman",
    "image": "https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333"
  }
]
