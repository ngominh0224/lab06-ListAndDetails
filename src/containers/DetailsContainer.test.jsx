import React from 'react';
import { render, screen } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import DetailsContainer from './DetailsContainer';
import mockData from './mockData.json';
import { MemoryRouter } from 'react-router-dom';
import { Route } from 'react-router-dom';

const server = setupServer(
  rest.get(
    'https://hey-arnold-api.herokuapp.com/api/v1/characters/5da237699734fdcb7bef8f51',
    (req, res, ctx) => {
      return res(ctx.json(mockData));
    }
  )
);

describe('DetailsContainer', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('gives one character', async () => {
    render(
      <MemoryRouter initialEntries={['/characters/5da237699734fdcb7bef8f51']}>
        <Route path="/characters/:id">
          <DetailsContainer />
        </Route>
      </MemoryRouter>
    );

    const unorderedList = await screen.findByRole('list', {
      name: 'characters',
    });
    expect(unorderedList).toMatchSnapshot();
  });
});
