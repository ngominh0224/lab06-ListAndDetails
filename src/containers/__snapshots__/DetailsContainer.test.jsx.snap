// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsContainer gives one character 1`] = `
<ul
  aria-label="characters"
>
  <li>
    <div>
      <a
        href="/characters/undefined"
      >
        <img />
        <p />
      </a>
    </div>
  </li>
</ul>
`;
